sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -977,6 +977,13 @@ class Analyzer(
         localPredicateReferences -- p.outputSet
       }
 
+      // SPARK-17348
+      // Report a non-supported case where there exists a correlated predicate
+      // that is not an equality predicate and the outer reference of the correlated predicate
+      // is not at the immediate parent operator.
+      def failOnNonEqualityPredicate(p: LogicalPlan): Unit = {
```

Comment:
yeah. I tried to make the error message reporting in line with existing ones, which have their own "def". I will collapse it in the main code.